processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategySelector.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.strategy;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.metamx.common.ISE;
+import io.druid.query.groupby.GroupByQuery;
+import io.druid.query.groupby.GroupByQueryConfig;
+import io.druid.query.groupby.epinephelinae.EpiGroupByStrategy;
+
+public class GroupByStrategySelector
+{
+  public static final String CTX_KEY_STRATEGY = "groupByStrategy";
+  public static final String STRATEGY_EPINEPHELINAE = "epinephelinae";
```

Comment:
while i understand and appreciate the analogy to fish, i m not sure if its just me, but i find "epinephelinae" hard to pronounce and remember spelling of... choosing "v1", "v2", "old/new" kind of string might be simpler.